qiskit_machine_learning/algorithms/classifiers/neural_network_classifier.py

```diff
@@ -174,4 +184,58 @@ def _validate_input(self, X: np.ndarray, y: np.ndarray = None) -> Tuple[np.ndarr
             elif scipy.sparse.issparse(y):
                 y = cast(spmatrix, y).toarray()  # cast is required by mypy
 
-        return X, y
+        return np.asarray(X), np.asarray(y)
+
+    def _validate_binary_targets(self, y: np.ndarray) -> None:
+        """Validate binary encoded targets.
+
+        Raises:
+            QiskitMachineLearningError: If targets are invalid.
+        """
+        if not len(y.shape) == 1 and len(np.unique(y)) == 2:
+            raise QiskitMachineLearningError(
+                f"Current settings only applicable to binary classification! Got labels: {y}"
+            )
+
+    def _validate_one_hot_targets(self, targets: np.ndarray) -> None:
+        """Validate one-hot encoded targets.
+
+        Ensure one-hot encoded data is valid and not multi-label.
+
+        Raises:
+            QiskitMachineLearningError: If targets are invalid.
+        """
+        if not np.isin(targets, [0, 1]).all():
```

Comment:
In this, and other error messages here,  the formatted value in the string is source from an numpy array. I am wondering if in general this is going to make these error messages rather large - depends on the array size and how numpy is set to format them I guess. If they are large by default numpy truncates the data which may end up not revealing the problem anyway. Here for instance can we not just say 'Invalid on-hot targets, data must contain only 0's and 1's' We could try and find some aspect of the data that fails and include that, I am just thinking the whole of the array is maybe too much.

Reply:
Yeah, good point. Those error messages weren't too helpful. I've edited them to, hopefully, make them clearer. Let me know if you think they still need some work.